pkg/core/plugins/interfaces.go

```diff
@@ -80,3 +83,10 @@ type AuthnAPIServerPlugin interface {
 	Plugin
 	NewAuthenticator(PluginContext) (authn.Authenticator, error)
 }
+
+// PolicyPlugin a plugin to add a Policy to Kuma
+type PolicyPlugin interface {
+	Plugin
+	MatchedPolicies(dataplane *core_mesh.DataplaneResource, resources xds_context.Resources) (core_xds.TypedMatchingPolicies, error)
```

Comment:
What about MeshGateway? Is it another plugin or we should think about a more generic proxy type as a first argument here?

Reply:
Isn't MeshGateway a dataplane in the end?

Reply:
You're right, we don't target MeshGateway for policies